packages/block-library/src/gallery/edit.js

```diff
@@ -177,6 +182,16 @@ class GalleryEdit extends Component {
 		} );
 	}
 
+	saveImageAttributes( id, attributes ) {
+		const data = new window.FormData();
+		forEach( attributes, ( ( value, key ) => data.append( key, value ) ) );
+		apiFetch( {
+			path: '/wp/v2/media/' + id,
+			body: data,
+			method: 'POST',
+		} );
+	}
```

Comment:
Why should we do so in the Gallery and not in the Image. I think the decision has been made in Gutenberg to not persist the caption into the Media Library because it's a confusing flow if you have the same image in multiple posts, but you want the caption to be different.

Reply:
Hey @youknowriad, when I worked on this and pushed this I wasn't aware of that decision, I was only trying to be helpful. I tried to read up on what others had posted about this and provide a possible solution that made sense to me. You can close this if you want.

Wherever that decision was made it seems to me like its a more confusion workflow with it not syncing with the media library. Especially when you edit the gallery and lose your captions. ¯\_(ツ)_/¯

Reply:
yeah, I added the "Needs Design Feedback" to get some opinions. I highly appreciate your willingness to contribute. I'm hoping we could find some ways to improve the flows there.